### DISCUSSION TITLE
Make module inclusions conditional

### DISCUSSION THREAD

**Discussion by mil-ad** [polybar/polybar#2940] [Q&A] [answered]
I have slightly different `modules-right`s on my laptop vs PC (e.g. my PC doesn't need the `battery` module) but ideally I'd like to use the same config file as I store it in my common dotfiles repository. 

Is there a way to support this in polybar? For instance by making the inclusion of a module conditional (e.g. based on the output of a shell script/command?). Or perhaps by adding some sort of `enable-if` flag to definition of the module itself. I came across the `include-file` and `include-directory` directives which are almost what I want but I think I also need some of sort of `if` statement?

**Comment by patrick96** [comment, marked as answer]
What I'm doing, is something like `modules-left = ${env:BAR_MODULES_TOP_LEFT}`. The environment is set in a bash script that runs before polybar starts. The script can populate the module list depending on the system it runs on.

I believe this system is much more flexible than any conditional constructs inside our config file.

**Comment by mil-ad** [comment]
very good, didn't know about the `${env:}` syntax
